Documentation for new setStatusMessage action and presenceStatusMessage type (#19239)
microsoftgraph/microsoft-graph-docs: api-reference/beta/resources/presenceStatusMessage.md

---
title: "presenceStatusMessage resource type"

> description: "Represents Teams Presence Status Message for a user's presence"

Reviewer: ```suggestion
description: Represents a presence status message for a user's presence on Teams".
```

author: "afedorov"
ms.localizationpriority: medium
ms.prod: "cloud-communications"
doc_type: resourcePageType
---

# presenceStatusMessage resource type

Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> Represents the Teams Presence Status Message related to the [presence](presence.md) of a user.

Reviewer: ```suggestion
Represents a presence status message related to the [presence](presence.md) of a user on Teams.
```

## Properties

| Property       | Type           | Description                                 | 
| -------------- | -------------- | ------------------------------------------- | 

> | message | [microsoft.graph.itemBody](itemBody.md) | Status message item.<br/><br/> The only supported format at the moment is `message.contentType = 'text'`. |

Reviewer: ```suggestion
| message | [microsoft.graph.itemBody](itemBody.md) | Status message item.<br/><br/> The only supported format currently is `message.contentType = 'text'`. |
```

> | publishedDateTime | Edm.DateTimeOffset | Moment in time of when status message was published. Read-only.<br/><br/>`publishedDateTime` is not available when requesting presence of another user. |

Reviewer 2: Can "Read-only" be in a new line for clarity

> | publishedDateTime | Edm.DateTimeOffset | Moment in time of when status message was published.<br/>Read-only.<br/><br/>`publishedDateTime` is not available when requesting presence of another user. |

Reviewer: ```suggestion
| publishedDateTime | Edm.DateTimeOffset |Time in which the status message was published.<br/>Read-only.<br/><br/>`publishedDateTime` is not available when requesting presence of another user. |
```

> | expiryDateTime | [microsoft.graph.dateTimeTimeZone](dateTimeTimeZone.md) | Moment in time of when status message expires.<br/>If not provided the status message will not expire.<br/><br/>`expiryDateTime.dateTime` should not include time zone.<br/><br/>`expiryDateTime` is not available when requesting presence of another user. |

Reviewer: ```suggestion
| expiryDateTime | [microsoft.graph.dateTimeTimeZone](dateTimeTimeZone.md) | Time in which the status message expires.<br/>If not provided, the status message does not expire.<br/><br/>`expiryDateTime.dateTime` should not include time zone.<br/><br/>`expiryDateTime` is not available when requesting presence of another user. |
```

> | expiryDateTime | [microsoft.graph.dateTimeTimeZone](dateTimeTimeZone.md) | Moment in time of when status message expires.<br/>If not provided the status message will not expire.<br/><br/>`expiryDateTime.dateTime` should not include time zone.<br/><br/>`expiryDateTime` is not available when requesting presence of another user. |

Reviewer: ```suggestion
| expiryDateTime | [microsoft.graph.dateTimeTimeZone](dateTimeTimeZone.md) | Time in which status message expires.<br/>If not provided, the status message does not expire.<br/><br/>`expiryDateTime.dateTime` should not include time zone.<br/><br/>`expiryDateTime` is not available when requesting presence of another user. |
```

> | expiryDateTime | [microsoft.graph.dateTimeTimeZone](dateTimeTimeZone.md) | Moment in time of when status message expires.<br/>If not provided the status message will not expire.<br/><br/>Please mind that `expiryDateTime.dateTime` should not include time zone.<br/><br/>`expiryDateTime.timeZone` may occur to be different from previously set value when using [Get presence](../api/presence-get.md) or [Get presence of multiple users](../api/cloudcommunications-getpresencesbyuserid.md) requests.<br/><br/>`expiryDateTime` is not available when requesting presence of another user. |

Reviewer 2: >Please **note** that `expiryDateTime.dateTime` should not include time zone

> | expiryDateTime | [microsoft.graph.dateTimeTimeZone](dateTimeTimeZone.md) | Moment in time of when status message expires.<br/>If not provided the status message will not expire.<br/><br/>Please mind that `expiryDateTime.dateTime` should not include time zone.<br/><br/>`expiryDateTime.timeZone` may occur to be different from previously set value when using [Get presence](../api/presence-get.md) or [Get presence of multiple users](../api/cloudcommunications-getpresencesbyuserid.md) requests.<br/><br/>`expiryDateTime` is not available when requesting presence of another user. |

Reviewer 2: `expiryDateTime.timeZone` may occur to be different from previously set value when using [Get presence](../api/presence-get.md) or [Get presence of multiple users](../api/cloudcommunications-getpresencesbyuserid.md) requests

This is confusing IMO. Any other ideas on making this more clear? Perhaps take Laura's help?

Author: @Lauragra could you please help to clarify this paragraph:
> `expiryDateTime.timeZone` may occur to be different from previously set value when using [Get presence](../api/presence-get.md) or [Get presence of multiple users](../api/cloudcommunications-getpresencesbyuserid.md) requests.

In presence-setstatusmessage.md we have:
> Status message text can be xml-sanitized which leads to having a different status message applied.

The thing is that the sanitization is performed by another service which we do not control.
They have a work item to clarify the sanitization mechanism in a different doc.

Meanwhile it would be surprising for our customers to discover this behaviour.
That's why I am trying to carefully mention sanitization.
Could you please help to make a proper text here?

Author: Please disregard the request above since we will remove any sanitization related text/examples and reconsider adding them when UPS team updates their docs.
